### PR TITLE
Move Merkle ledger tests to Alcotest

### PR DESCRIPTION
**Pull request by rbonichon**
Use `alcotest` to express tests for Merkle ledger(s).

Sneaked in a small update in one the test Modules.

### CODE REVIEW

**Comment by rbonichon** [comment]
!ci-build-me

**Review by svv232 — COMMENTED** [review]
the refactor logic makes sense to me, but i think this is breaking dev unit tests right now.    The syncable ledger tests seem to reference `merkle_ledger_tests`.

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by rbonichon** [comment]
!ci-build-me

File: src/lib/merkle_ledger_tests/test_mask.ml [hunk 1]
```diff
@@ -156,477 +172,505 @@ module Make (Test : Test_intf) = struct
...
+              List.map2_exn account_ids balances ~f:Account.create
+            in
+            (* add accounts to parent *)
+            List.iter parent_accounts ~f:(fun account ->
+                ignore @@ parent_create_new_account_exn maskable account ) ;
+            let balance_summer _addr accum acct =
+              accum + Balance.to_nanomina_int (Account.balance acct)
+            in
+            let parent_sum =
+              Maskable.foldi maskable ~init:0 ~f:balance_summer
```
Comment (georgeee) [inline src/lib/merkle_ledger_tests/test_mask.ml:546]:
```suggestion
              Maskable.fold maskable ~init:0 ~f:balance_summer
```

and remove unused parameter from the `balance_summer` function (same modification could be applied below)

File: src/lib/merkle_ledger_tests/test_mask.ml [hunk 2]
```diff
@@ -156,477 +172,505 @@ module Make (Test : Test_intf) = struct
...
+          Test.with_chain (fun _ ~mask:mask1 ~mask_as_base:_ ~mask2 ->
+              let num_accounts = 1 lsl Test.depth in
+              let gen_values gen list_length =
+                Quickcheck.random_value
+                  (Quickcheck.Generator.list_with_length list_length gen)
+              in
+              let account_ids = Account_id.gen_accounts num_accounts in
+              let balances = gen_values Balance.gen num_accounts in
+              let base_accounts =
+                List.map2_exn account_ids balances ~f:(fun public_key balance ->
```
Comment (georgeee) [inline src/lib/merkle_ledger_tests/test_mask.ml:401]:
```suggestion
                List.map2_exn account_ids balances ~f:Account.create
```

File: src/lib/merkle_ledger_tests/test_mask.ml [hunk 3]
```diff
@@ -156,477 +172,505 @@ module Make (Test : Test_intf) = struct
     in
     test_hashes_at_address addr
 
-  let%test "parent, mask agree on set" =
```
Comment (georgeee) [inline src/lib/merkle_ledger_tests/test_mask.ml:159]: Do I understand correctly that this test was written twice (same code, duplicated)?

**Review by georgeee — APPROVED** [review]

**Review by georgeee — APPROVED** [review]

File: src/lib/merkle_ledger_tests/test_stubs.ml [hunk 4]
```diff
@@ -59,12 +59,12 @@ module Hash = struct
    * important impossible to create an account such that (merge a b = hash_account account) *)
 
   let hash_account account =
-    Md5.digest_string ("0" ^ Format.sprintf !"%{sexp: Account.t}" account)
+    Md5.digest_string (Format.sprintf !"0%{sexp: Account.t}" account)
```
[the remark this answers is not in the document]
Comment (rbonichon) [reply src/lib/merkle_ledger_tests/test_stubs.ml:62]: Kind of. This is also refactoring, isn't it?.

File: src/lib/merkle_ledger_tests/test_mask.ml (hunk 3 above)
Comment (rbonichon) [reply src/lib/merkle_ledger_tests/test_mask.ml:159]: Exactly

File: src/lib/merkle_ledger_tests/test_mask.ml (hunk 1 above)
Comment (rbonichon) [reply src/lib/merkle_ledger_tests/test_mask.ml:546]: unfortunately `fold` does not exist.

File: src/lib/merkle_ledger_tests/test_mask.ml [hunk 5]
```diff
@@ -156,477 +172,505 @@ module Make (Test : Test_intf) = struct
...
+                List.iter locs ~f:(fun (loc, a) ->
+                    [%test_result: Account.t option]
+                      ~message:"All accounts are accessible from m2"
+                      ~expect:(Some a) (Mask.Attached.get m2 loc) )
+            | _ ->
+                failwith "unexpected" ) )
+
+  let () =
+    add_test
+      "set account in parent doesn't remove masked copy if mask is still dirty \
```
[the remark this answers is not in the document]
Comment (rbonichon) [reply src/lib/merkle_ledger_tests/test_mask.ml]: Done

File: src/lib/merkle_ledger_tests/test_stubs.ml (hunk 4 above)
[the remark this answers is not in the document]
Comment (georgeee) [reply src/lib/merkle_ledger_tests/test_stubs.ml:62]: it's alright to keep it as is

**Pull request merged**